cmd/service_bootstrap_desktop.go

```diff
@@ -0,0 +1,120 @@
+// +build darwin windows linux
+
+/*
+ * Copyright (C) 2018 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package cmd
+
+import (
+	"github.com/mysteriumnetwork/node/communication"
+	nats_dialog "github.com/mysteriumnetwork/node/communication/nats/dialog"
+	nats_discovery "github.com/mysteriumnetwork/node/communication/nats/discovery"
+	"github.com/mysteriumnetwork/node/core/node"
+	promise_noop "github.com/mysteriumnetwork/node/core/promise/methods/noop"
+	"github.com/mysteriumnetwork/node/core/service"
+	"github.com/mysteriumnetwork/node/discovery"
+	"github.com/mysteriumnetwork/node/identity"
+	identity_selector "github.com/mysteriumnetwork/node/identity/selector"
+	dto_discovery "github.com/mysteriumnetwork/node/service_discovery/dto"
+	service_noop "github.com/mysteriumnetwork/node/services/noop"
+	service_openvpn "github.com/mysteriumnetwork/node/services/openvpn"
+	openvpn_service "github.com/mysteriumnetwork/node/services/openvpn/service"
+	service_wireguard "github.com/mysteriumnetwork/node/services/wireguard"
+	"github.com/mysteriumnetwork/node/session"
+)
+
+func (di *Dependencies) bootstrapServices(nodeOptions node.Options) {
```

Comment:
I like this small function which is really desktop specific. So a suggestion - do component bootstraping as part of common code. Extract bootStrapServices method which might return erorr. On desktop - bootstrap openvpn, noop, wireguard, on mobile return error. Any thoughts? You will avoid if runner == nil check and return of the error by doing compile time decision return error or not.

Reply:
excellent idea!